apps/bfd-server/bfd-server-war/src/main/java/gov/cms/bfd/server/war/r4/providers/preadj/R4ClaimSamhsaMatcher.java

```diff
@@ -28,4 +37,33 @@ public boolean test(Claim claim) {
         || containsSamhsaIcdDiagnosisCode(adapter.getDiagnosis())
         || containsSamhsaLineItem(adapter.getItem());
   }
+
+  /**
+   * Checks if the given {@link CodeableConcept} contains only known coding systems.
+   *
+   * <p>For v2 FHIR resources, we include backwards compatability for the
+   * https://bluebutton.cms.gov/resources/variables/hcpcs_cd system.
+   *
+   * @param procedureConcept the procedure {@link CodeableConcept} to check
+   * @return <code>true</code> if the specified procedure {@link CodeableConcept} contains at least
+   *     one {@link Coding} and only contains {@link Coding}s that have known coding systems, <code>
+   *     false</code> otherwise
+   */
+  @Override
+  protected boolean containsOnlyKnownSystems(CodeableConcept procedureConcept) {
+    Set<String> codingSystems =
+        procedureConcept.getCoding().stream().map(Coding::getSystem).collect(Collectors.toSet());
+
+    return codingSystems.equals(HCPCS_SYSTEM);
+  }
```

Comment:
Making an abstract method that simply returned a collection of system sets was the first approach I took, but it seemed clumsy and use case specific, as EOB is really the only one that compares that way.  Partially adjudicated claims want one system and one system only, so it seemed more sensible to implement this method.  The stream call is duplicated, but it's sensible for it to exist in this method, since other methods don't need knowledge of the innerworkings of this procedure concept.

Reply:
This could be written a different way which might be more direct to what its intent is:
```
return procedureConcept.getCoding().stream()
    .noneMatch(c -> c.getSystem.equals(TransformerConstants.CODING_SYSTEM_CARIN_HCPCS));